### DISCUSSION TITLE
StorageConverter: can a started storagemanager configuration be used as a source?

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#622] [Q&A] [answered]
Hi all;
using StorageConverter class directly, we need to pass 2 input params: source configuration and target configuration.
for the source config,  can I send a configuration of a running MS storage (via embeddedStorageManager.configuration() ) ?
or is this error prone because its running (started).
I have checked both with a running storage and a non running and results seem to be the same but I am trying to get a definite answer.
thanks

**Comment by hg-ms** [comment, marked as answer]
You can use the storage configuration from a running storage, but you should ensure that the storage is no more running during the conversion as it is not guaranteed that the source storage won’t be altered by the housekeeping. In the worst case that may cause a corrupted target storage.

**Reply by guykatz** [reply]
thanks @hg-ms ill make sure its shutdown to avoid issues...
